src/index.ts

```diff
@@ -1054,11 +1067,50 @@ export class Profile {
     return offset
   }
 
+  _encodeToBuffer(buffer: Uint8Array, offset = 0): number {
+    offset = this._encodeSampleTypesToBuffer(buffer, offset)
+    offset = this._encodeSamplesToBuffer(buffer, offset)
+    offset = this._encodeMappingsToBuffer(buffer, offset)
+    offset = this._encodeLocationsToBuffer(buffer, offset)
+    offset = this._encodeFunctionsToBuffer(buffer, offset)
+    offset = this.stringTable._encodeToBuffer(buffer, offset)
+    offset = this._encodeBasicValuesToBuffer(buffer, offset)
+    return offset
+  }
+
+  async _encodeToBufferAsync(buffer: Uint8Array, offset = 0): Promise<number> {
+    offset = this._encodeSampleTypesToBuffer(buffer, offset)
+    await new Promise(setImmediate)
```

Comment:
So _that's_ how you do `Thread.yield()` equivalent in Node! :-)

Reply:
The `setImmediate` could also be `queueMicrotask`, but it's less greedy this way, which is better for the latency-avoidance purpose. 😅